File: Estudo-API-Casa-Do-Codigo/src/main/java/io/github/monthalcantara/casadocodigo/repository/AutorRepository.java
package io.github.monthalcantara.casadocodigo.repository;

import io.github.monthalcantara.casadocodigo.domain.Autor;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface AutorRepository extends JpaRepository<Autor, Long> {
}

File: Kotlin/API-Casa_Do_Codigo_Kotlin/src/main/kotlin/io/github/monthalcantara/casadocodigo/repository/AutorRepository.kt
package io.github.monthalcantara.casadocodigo.repository

import io.github.monthalcantara.casadocodigo.model.Autor
import org.springframework.data.jpa.repository.JpaRepository

interface AutorRepository : JpaRepository<Autor, Long> {
    fun save(entity: Autor): Autor
    fun existsByEmail(email: String): Boolean
}